File: Java/Technology Fundamentals with Java/Exercises Java Code/src/MethodExercise/PasswordValidator.java
package MethodExercise;

import java.util.Scanner;
import java.util.regex.Pattern;

public class PasswordValidator {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String password = sc.nextLine();

        boolean lengthInRange = isPasswordLengthTrue(password);
        boolean contentRestriction = isPasswordConsistOnlyDigitsAndLetters(password);
        boolean consistMinCountDigits = isPasswordConsistMinCountDigits(password);

        if (lengthInRange && contentRestriction && consistMinCountDigits) {
            System.out.println("Password is valid");
        }
        if (!lengthInRange) {
            System.out.println("Password must be between 6 and 10 characters");
        }

        if (!contentRestriction) {
            System.out.println("Password must consist only of letters and digits");
        }

        if (!consistMinCountDigits) {
            System.out.println("Password must have at least 2 digits");
        }
    }

    private static boolean isPasswordConsistMinCountDigits(String password) {
        int count = 0;
        for (int i = 0; i < password.length(); i++) {
            if (Pattern.matches("[0-9]+", password.charAt(i) + "")) {
                count++;
            }
        }
        if (count >= 2) {
            return true;
        }
        return false;
    }

    static boolean isPasswordLengthTrue(String password) {
        boolean isTrue = (password.length() >= 6 && password.length() <= 10) ? true : false;
        return isTrue;
    }

    static boolean isPasswordConsistOnlyDigitsAndLetters(String password) {
        if (password.matches("[a-zA-z0-9]+")) {
            return true;
        }
        return false;
    }

    static boolean consistOnlyDigitsAndLetters(String password) {
        boolean isValid = true;

        for (int i = 0; i < password.length(); i++) {
            if (!Character.isLetterOrDigit(password.charAt(i))) {
                isValid = false;
                break;
            }
        }
        return isValid;
    }
}

File: C#/Technology Fundamentals with C#/04. Methods - Functions - Exercise/04. Password Validator/PasswordValidator.cs
using System;

public class PasswordValidator
{
    public static void Main()
    {
        string password = Console.ReadLine();
        bool isBetween6And10Symbols = CheckLenghtOfPassword(password);
        if (isBetween6And10Symbols == false)
        {
            Console.WriteLine("Password must be between 6 and 10 characters");
        }
        bool containsOnlyDigitsLetters = containsOnlyDigitsAndLetters(password);

        if (containsOnlyDigitsLetters == false)
        {
            Console.WriteLine("Password must consist only of letters and digits");
        }

        bool containsMin2Digits = CheckMinDigit(password);
        if (containsMin2Digits == false)
        {
            Console.WriteLine("Password must have at least 2 digits");
        }

        if (isBetween6And10Symbols && containsOnlyDigitsLetters && containsMin2Digits)
        {
            Console.WriteLine("Password is valid");
        }
    }

    private static bool CheckMinDigit(string password)
    {
        int count = 0;
        for (int i = 0; i < password.Length; i++)
        {
            char symbol = password[i];
            if (char.IsDigit(symbol))
            {
                count++;
            }
        }
        return count >= 2 ? true : false;
    }

    private static bool containsOnlyDigitsAndLetters(string password)
    {
        for (int i = 0; i < password.Length; i++)
        {
            char symbol = password[i];
            if (!char.IsLetterOrDigit(symbol))
            {
                return false;
            }
        }
        return true;
    }

    private static bool CheckLenghtOfPassword(string password)
    {
        return password.Length >= 6 && password.Length <= 10 ? true : false;

    }
}
